perfdash/perfdash.go

```diff
@@ -81,12 +87,23 @@ func run() error {
 		options.DefaultBuildsCount = maxBuilds
 	}
 
-	gcs, err := NewGCSMetricsBucket(*logsBucket, *logsPath, *credentialPath)
+	var metricsBucket MetricsBucket
+	var err error
+
+	switch options.Mode {
+	case gcsMode:
+		metricsBucket, err = NewGCSMetricsBucket(*logsBucket, *logsPath, *credentialPath)
+	case s3Mode:
+		metricsBucket, err = NewS3MetricsBucket(*logsBucket, *logsPath, *awsRegion)
+	default:
+		return fmt.Errorf("unexpected mode: %s", options.Mode)
+	}
+
 	if err != nil {
 		panic(err)
```

Comment:
since you touch this code - can you please return the error here, instead of panic-ing?

Reply:
Misunderstood your earlier comment about this. I think I got it right this time hopefully, please take a look :)